client/data/emails/use-add-email-forward-mutation.ts

```diff
@@ -0,0 +1,194 @@
+import { useTranslate } from 'i18n-calypso';
+import { orderBy } from 'lodash';
+import { createElement } from 'react';
+import { useIsMutating, useMutation, useQueryClient } from 'react-query';
+import { useDispatch, useSelector } from 'react-redux';
+import { getCacheKey as getEmailDomainsQueryKey } from 'calypso/data/domains/use-get-domains-query';
+import { CALYPSO_CONTACT } from 'calypso/lib/url/support';
+import wp from 'calypso/lib/wp';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { getCacheKey as getEmailAccountsQueryKey } from './use-get-email-accounts-query';
+import type { UseMutationOptions } from 'react-query';
+
+type AddMailboxFormData = {
+	destination: string;
+	mailbox: string;
+};
+
+type Context = {
+	[ key: string ]: any;
+};
+
+const MUTATION_KEY = 'addEmailForward';
+
+export function useIsLoading() {
```

Comment:
`useMutation` does return an `isLoading` flag that I initially tried to use, but unfortunately, that flag is [not preserved across components](https://github.com/TanStack/query/issues/2304).

This is a way of working around that.